compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -260,6 +265,218 @@ static CodeGenPipeline getTensorCorePipeline(Type elementType) {
+  SmallVector<int64_t> mapping(op.getNumLoops());
+  std::iota(mapping.begin(), mapping.end(), 0);
+
+  int64_t lastReductionDim = reductionDims.back();
+
+  for (int64_t dim : parallelDims) {
+    workgroupTileSizes[dim] = 1;
+  }
+
+  for (int64_t dim : reductionDims) {
+    threadTileSizes[dim] = 1;
+    partialReductionTileSizes[dim] = 1;
+  }
+
+  int64_t reductionSize = 1;
+  for (int64_t dim : reductionDims)
+    reductionSize *= bounds[dim];
+
+  int64_t subgroupSize = 0;
+  for (int s : wgp.getSubgroupSizeChoices().asArrayRef()) {
+    if (reductionSize % s == 0) {
+      subgroupSize = s;
+      break;
+    }
+  }
+  if (subgroupSize == 0)
+    return failure();
+
+  auto bitWidth = getBitWidth(op);
+  if (failed(bitWidth)) {
+    return failure();
+  }
+
+  const std::optional<int64_t> maxLoadBits = wgp.getMaxLoadInstructionBits();
+  const unsigned largestLoadSizeInBits =
+      maxLoadBits.has_value() ? *maxLoadBits : 128;
+
+  int64_t threadLoads = largestLoadSizeInBits / *bitWidth;
+  if (reductionSize % threadLoads != 0)
+    return failure();
+
+  int64_t lastDimSize = bounds[reductionDims.back()];
+  int64_t partialReductionSize = subgroupSize * threadLoads;
+
+  while (lastDimSize % partialReductionSize != 0)
+    partialReductionSize >>= 1;
+
+  threadLoads = std::min(lastDimSize, threadLoads);
+
+  int64_t threadBasis = partialReductionSize / threadLoads;
+
+  partialReductionTileSizes[lastReductionDim] = partialReductionSize;
+  threadTileSizes[lastReductionDim] = threadLoads;
+  threadCounts[lastReductionDim] = threadBasis;
+
+  auto context = op.getContext();
+  Builder b(context);
+  SmallVector<NamedAttribute, 1> attrs;
+
+  ArrayAttr subgroupBasisAttr = b.getArrayAttr(
+      {b.getI64ArrayAttr(subGroupCounts), b.getI64ArrayAttr(mapping)});
+
+  ArrayAttr threadBasisAttr = b.getArrayAttr(
+      {b.getI64ArrayAttr(threadCounts), b.getI64ArrayAttr(mapping)});
+
+  attrs.emplace_back(b.getStringAttr("workgroup"),
+                     b.getI64ArrayAttr(workgroupTileSizes));
+  attrs.emplace_back(b.getStringAttr("partial_reduction"),
+                     b.getI64ArrayAttr(partialReductionTileSizes));
+  attrs.emplace_back(b.getStringAttr("thread"),
+                     b.getI64ArrayAttr(threadTileSizes));
+  attrs.emplace_back(b.getNamedAttr("thread_basis", threadBasisAttr));
+  attrs.emplace_back(b.getNamedAttr("subgroup_basis", subgroupBasisAttr));
+  auto configDict = b.getDictionaryAttr(attrs);
+  auto loweringConfig = IREE::GPU::LoweringConfigAttr::get(context, configDict);
+
+  SmallVector<NamedAttribute, 1> pipelineAttrs;
+  auto pipelineOptions = IREE::GPU::GPUPipelineOptionsAttr::get(
+      context, /*prefetchSharedMemory=*/false,
+      /*no_reduce_shared_memory_bank_conflicts=*/true,
+      /*use_igemm_convolution=*/false,
+      /*reorder_workgroups_strategy=*/std::nullopt);
+
+  pipelineAttrs.emplace_back(
+      b.getStringAttr(IREE::GPU::GPUPipelineOptionsAttr::getDictKeyName()),
+      pipelineOptions);
```

Comment:
Similar here: instead of declaring this vector a few lines above and then adding one element here, can't we declare this argument as a local variable passed to `getDictionaryAttr`? I guess it needs an `ArrayRef` and it does have a constructor for a single argument